db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -27,68 +31,105 @@ def request_totals(template_totals, dialog_totals)
+      service_prov_option_value(prov_option, child_service_resource.resource, options_array)
     end
   end
   options_array
 end
 
+def service_prov_option_value(prov_option, resource, options_array = [])
+  args_hash = {:prov_option   => prov_option,
+               :array         => options_array,
+               :resource      => resource,
+               :flavor        => flavor_obj(resource.get_option(:instance_type)),
+               :number_of_vms => get_option_value(resource, :number_of_vms),
+               :cloud         => cloud?(resource.get_option(:st_prov_type))}
+
+  case prov_option
+  when :vm_memory
+    args_hash[:prov_value] = args_hash[:number_of_vms] * get_option_value(resource, :vm_memory)
+    request_hash_value(args_hash)
+  when :number_of_cpus
+    requested_number_of_cpus(args_hash)
+  when :storage
+    requested_storage(args_hash)
+  else
+    options_value(args_hash)
+  end
+  options_array
+end
+
+def vm_provision_cloud?
+  @miq_request.source.respond_to?(:cloud) ? @miq_request.source.cloud : false
+end
+
+def flavor_obj(id)
+  vmdb_object(:flavor, id)
+end
+
 def vm_prov_option_value(prov_option, options_array = [])
-  number_of_vms = get_option_value(@miq_request, :number_of_vms)
+  args_hash = {:prov_option   => prov_option,
+               :array         => options_array,
+               :resource      => @miq_request,
+               :flavor        => flavor_obj(@miq_request.get_option(:instance_type)),
+               :number_of_vms => get_option_value(@miq_request, :number_of_vms),
+               :cloud         => vm_provision_cloud?}
+
   case prov_option
   when :vm_memory
-    memory_in_request = number_of_vms * get_option_value(@miq_request, :vm_memory)
-    set_requested_value(prov_option, memory_in_request,
-                        @miq_request.get_option(:instance_type), options_array)
+    args_hash[:prov_value] = args_hash[:number_of_vms] * get_option_value(@miq_request, :vm_memory)
+    request_hash_value(args_hash)
   when :number_of_cpus
-    requested_number_of_cpus(number_of_vms, options_array)
+    requested_number_of_cpus(args_hash)
   when :storage
-    requested_storage(prov_option, number_of_vms, options_array)
+    requested_storage(args_hash)
   else
-    options_value(prov_option, @miq_request, options_array)
+    options_value(args_hash)
   end
   options_array
 end
 
-def requested_number_of_cpus(number_of_vms, options_array)
-  cpu_in_request = get_option_value(@miq_request, :number_of_cpus)
+def requested_number_of_cpus(args_hash)
+  cpu_in_request = get_option_value(args_hash[:resource], args_hash[:number_of_cpus])
   if cpu_in_request.zero?
-    cpu_in_request = get_option_value(@miq_request, :number_of_sockets) *
-                     get_option_value(@miq_request, :cores_per_socket)
+    cpu_in_request = get_option_value(args_hash[:resource], :number_of_sockets) *
+                     get_option_value(args_hash[:resource], :cores_per_socket)
```

Comment:
Make this the default since it is what the current dialogs use then fall back to `number_of_cpus` if the result is zero.